src/MiddlewareListener.php

```diff
@@ -6,38 +6,46 @@
  * @license   https://github.com/laminas/laminas-mvc-middleware/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Laminas\Mvc\Middleware;
 
-use Exception;
-use Interop\Http\ServerMiddleware\MiddlewareInterface;
+use Closure;
 use Laminas\EventManager\AbstractListenerAggregate;
 use Laminas\EventManager\EventManagerInterface;
 use Laminas\Http\Response;
 use Laminas\Mvc\Application;
 use Laminas\Mvc\Exception\InvalidMiddlewareException;
 use Laminas\Mvc\MvcEvent;
 use Laminas\Psr7Bridge\Psr7Response;
+use Laminas\Stratigility\Middleware\CallableMiddlewareDecorator;
+use Laminas\Stratigility\Middleware\RequestHandlerMiddleware;
 use Laminas\Stratigility\MiddlewarePipe;
 use Psr\Container\ContainerInterface;
 use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
 use Throwable;
 
+use function get_class;
+use function gettype;
+use function is_object;
+use function is_string;
+
 class MiddlewareListener extends AbstractListenerAggregate
 {
     /**
-     * Attach listeners to an event manager
-     *
-     * @return void
+     * {@inheritDoc}
      */
-    public function attach(EventManagerInterface $events, $priority = 1)
+    public function attach(EventManagerInterface $events, $priority = 1) : void
     {
         $this->listeners[] = $events->attach(MvcEvent::EVENT_DISPATCH, [$this, 'onDispatch'], 1);
     }
 
     /**
      * Listen to the "dispatch" event
      *
-     * @return null|Response
+     * @return null|Response|mixed
```

Comment:
I'd replace this with just "mixed"; psalm is going to tell you to do that anyways.